src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -648,7 +648,7 @@ static void addERR_ParamsCantBeWithModifier(BindingDiagnosticBag diagnostics, Sy
                 diagnostics.Add(ErrorCode.ERR_MethodArgCantBeRefAny, parameterSyntax.Location, parameter.Type);
             }
 
-            if (parameter.Scope == DeclarationScope.ValueScoped && !parameter.Type.IsErrorTypeOrRefLikeType())
+            if (parameter.Scope == DeclarationScope.ValueScoped && !parameter.TypeWithAnnotations.IsRefLikeType())
```

Comment:
Is there a way to also check for the error type case without pulling too aggressively on lazy `.Type`?

Reply:
Consider adding a comment at the top of this method that pulling on `parameter.Type` is cycle-prone.

Reply:
> Is there a way to also check for the error type case without pulling too aggressively on lazy `.Type`?

We could add a similar `IsErrorType()` method to `TypeWithAnnotations`, but I'm not sure it's worth it. (Should an unresolved type always return `false` for `IsErrorType()`?)

I've added a comment to the method to avoid using `parameter.Type`.